physics/continuous_trajectory.hpp

```diff
@@ -193,6 +194,19 @@ class ContinuousTrajectory : public Trajectory<Frame> {
   // The polynomials are in increasing time order.
   InstantPolynomialPairs polynomials_;
 
+  // Lookups into |polynomials_| are expensive because they entail a binary
+  // search into a vector that grows over time.  In benchmarks, this can be as
+  // costly as the polynomial evaluation itself.  The accesses are not random,
+  // though, they are clustered in time and (slowly) increasing.  To take
+  // advantage of this, we keep track of the index of the last accessed
+  // polynomial and first try to see if the new lookup is for the same
+  // polynomial.  This makes us O(1) instead of O(Log N) most of the time and it
+  // speeds up the lookup by a factor of 7.  This member is atomic because of
+  // multithreading in the ephemeris, and is mutable to maintain the fiction
+  // that evaluation has no side effects.  Any value in the range of
+  // |polynomials_| or 0 is correct.
```

Comment:
Comment on the alternative of a `std::map<thread::id, int>`.

Reply:
I am explaining why it works with multithreading.  No point in documenting the roads not taken.

Reply:
For the record: we are trying to optimize a binary search.  We know that if `polynomials_` is a map the performance is bad.  So surely adding a map lookup to try to optimize a binary search doesn't make sense.